x/group/keeper/msg_server.go

```diff
@@ -847,11 +823,13 @@ func (k Keeper) LeaveGroup(goCtx context.Context, req *group.MsgLeaveGroup) (*gr
 
 	// update group weight
 	groupInfo.TotalWeight = updatedWeight.String()
-	if err := k.groupTable.Update(ctx.KVStore(k.key), groupInfo.Id, &groupInfo); err != nil {
+	groupInfo.Version++
```

Comment:
vulnerability 3: we forgot to bump version (though it's benign, because version is not used for checking proposals anymore).